projection/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -427,7 +417,7 @@ private[projection] class R2dbcOffsetStore(
         } else
           newState
 
-      R2dbcExecutor.updateInTx(stmts).map { _ =>
+      R2dbcExecutor.updateBatchInTx(batchStatement).map { _ =>
```

Comment:
small optimization could be to use `updateOneInTx` when there is only one record (which is the normal case for exactly-once)

Reply:
like this 74c1f26 ?